### ISSUE TITLE
react-tap-event-plugin requires ^2.x, but React and ReactDOM ^15.x supported - these are not compatible

### ISSUE THREAD

**Issue by marcacyr** [Sharlaan/material-ui-superselectfield#18]
This is a simple one and a suggestion (see comment as it might be completely unnecessary as I've been inundated and haven't been able to dig into all the upgrades I've wanted to do to our packages):

react-tap-event-plugin ^2.x is exclusively for react and react-dom (addons etc) at ^15.4.2, and is 100% incompatible with react 15.3.2 (the prior version).

So, by accepting react at 15.x, the range for react-tap-event-plugin should be 1.x to 2.x, as it was before when it was `"^1.x || ^2.x"`

The solution (in my opinion) is NOT to enforce react 15.4.2 and higher because not everyone is ready to make the jump quite yet. Jumping to 15.4.2 requires bumping Material UI to a higher version than 16.1.0, but they are having significant issues with higher versions right now. Specifically, some of their onTouchTaps are not working, and things like dropdowns are broken. We discovered this when trying to do an upgrade last week to higher versions.

So, what should happen is: 1) continue to allow for a wide range of React 15.x and 2) go back to allowing a range of react-tap-event-plugin so 1.x and 2.x are allowed.

I am currently using 1.5.1 of this package with react-tap-event-plugin at 1.0.0 and everything works really well! So, I know from first-hand knowledge consuming the package that other than the "warning" there is no issue!

Basically.. this would just get rid of the warning for people not ready to upgrade that one package.

**Comment by marcacyr** [comment]
Although... thinking about it a bit more, I think one can bump up to 15.4.2 and 2.0.x and use 16.1.0 MUI and 1.5.1 of this package and be fine.... I don't have time to test that out right now... so maybe this is a non-issue. Apologies if it is.

Thank you a thousand times @Sharlaan for being so, so responsive and republishing the package!!!

**Comment by Sharlaan** [comment]
oh nice infos you gave there !
ill revert this change back to `"^1.x || ^2.x"`

Thanks again for the report !

**Comment by marcacyr** [comment]
Sure thing! Thank you!
